backend/local/backend_apply.go

```diff
@@ -28,12 +29,22 @@ func (b *Local) opApply(
 	b.ContextOpts.Hooks = append(b.ContextOpts.Hooks, countHook, stateHook)
 
 	// Get our context
-	tfCtx, state, err := b.context(op)
+	tfCtx, opState, err := b.context(op)
 	if err != nil {
 		runningOp.Err = err
 		return
 	}
 
+	// context acquired the state, and therefor the lock.
+	// Unlock it when the operation is complete
+	defer func() {
+		if s, ok := opState.(state.Locker); op.LockState && ok {
+			if err := s.Unlock(); err != nil {
+				log.Printf("[ERROR]: %s", err)
```

Comment:
We should multierror append the error to `runningOp.Err` so that the error shows up to the end user. I would make a long const (like other error messages in the package) for it so that the user knows what to do: verify everything is okay, manually call `terraform unlock`